providers/mlx5/mlx5.c

```diff
@@ -609,16 +610,35 @@ static int mlx5_map_internal_clock(struct mlx5_device *mdev,
 	return 0;
 }
 
+struct ibv_cq_ex *mlx5dv_create_cq_ex(struct ibv_context *context,
+				      struct ibv_cq_init_attr_ex *cq_attr,
+				      struct mlx5dv_cq_init_attr_ex *mlx5_cq_attr)
+{
+	return ibv_vendor_create_cq_ex(context, cq_attr, mlx5_cq_attr);
+}
```

Comment:
Looks like you missed the point here, without calling libibverbs it won't have the chance to activate any private API upon this flow, currently there is some pre and post stuff that is done in libibverbs, the idea is that all of this code won't be duplicated to all providers, in addition any future support that may be added as of resource tracking per context will be in one place as well.

By using that wrapper the input private attr (i.e mlx5_cq_attr) is passed all the way from application to the driver without any need to save it locally on some context and the shared code of libibverbs is used, as of create CQ is not considered to be a performance path we see it as the  preferred way to go.